cli/command/defaultcontextstore_test.go

```diff
@@ -0,0 +1,223 @@
+	cli, err := NewDockerCli()
+	assert.NilError(t, err)
+	defer env.Patch(t, "DOCKER_HOST", "ssh://someswarmserver")()
+	defer env.Patch(t, "KUBECONFIG", "./testdata/test-kubeconfig")()
+	cli.configFile = &configfile.ConfigFile{
+		StackOrchestrator: "all",
+	}
+	ctx, err := resolveDefaultContext(&cliflags.CommonOptions{
+		TLS: true,
+		TLSOptions: &tlsconfig.Options{
+			CAFile: "./testdata/ca.pem",
+		},
+	}, cli.ConfigFile(), cli.Err())
+	assert.NilError(t, err)
+	assert.Equal(t, "default", ctx.Meta.Name)
+	assert.Equal(t, OrchestratorAll, ctx.Meta.Metadata.(DockerContext).StackOrchestrator)
+	assert.DeepEqual(t, "ssh://someswarmserver", ctx.Meta.Endpoints[docker.DockerEndpoint].(docker.EndpointMeta).Host)
+	golden.Assert(t, string(ctx.TLS.Endpoints[docker.DockerEndpoint].Files["ca.pem"]), "ca.pem")
+	assert.DeepEqual(t, "zoinx", ctx.Meta.Endpoints[kubernetes.KubernetesEndpoint].(kubernetes.EndpointMeta).DefaultNamespace)
+}
+
+func TestExportDefaultImport(t *testing.T) {
+	testDir, err := ioutil.TempDir("", t.Name())
+	assert.NilError(t, err)
+	defer os.RemoveAll(testDir)
+	dirStore := store.New(testDir, testCfg)
+	file1 := make([]byte, 1500)
+	rand.Read(file1)
+	file2 := make([]byte, 3700)
+	rand.Read(file2)
+	s := &ContextStoreWithDefault{
+		Store: dirStore,
+		Resolver: func() (*DefaultContext, error) {
+			return &DefaultContext{
+				Meta: testDefaultMetadata(),
+				TLS: store.ContextTLSData{
+					Endpoints: map[string]store.EndpointTLSData{
+						"ep2": {
+							Files: map[string][]byte{
+								"file1": file1,
+								"file2": file2,
+							},
+						},
+					},
+				},
+			}, nil
+		},
+	}
+	r := store.Export("default", s)
+	defer r.Close()
+	err = store.Import("dest", s, r)
+	assert.NilError(t, err)
+	srcMeta, err := s.GetContextMetadata("default")
+	assert.NilError(t, err)
+	destMeta, err := s.GetContextMetadata("dest")
+	assert.NilError(t, err)
+	assert.DeepEqual(t, destMeta.Metadata, srcMeta.Metadata)
+	assert.DeepEqual(t, destMeta.Endpoints, srcMeta.Endpoints)
+	srcFileList, err := s.ListContextTLSFiles("default")
+	assert.NilError(t, err)
+	destFileList, err := s.ListContextTLSFiles("dest")
+	assert.NilError(t, err)
+	assert.Equal(t, 1, len(destFileList))
+	assert.Equal(t, 1, len(srcFileList))
+	assert.Equal(t, 2, len(destFileList["ep2"]))
+	assert.Equal(t, 2, len(srcFileList["ep2"]))
+	srcData1, err := s.GetContextTLSData("default", "ep2", "file1")
+	assert.NilError(t, err)
+	assert.DeepEqual(t, file1, srcData1)
+	srcData2, err := s.GetContextTLSData("default", "ep2", "file2")
+	assert.NilError(t, err)
+	assert.DeepEqual(t, file2, srcData2)
+	destData1, err := s.GetContextTLSData("dest", "ep2", "file1")
+	assert.NilError(t, err)
+	assert.DeepEqual(t, file1, destData1)
+	destData2, err := s.GetContextTLSData("dest", "ep2", "file2")
+	assert.NilError(t, err)
+	assert.DeepEqual(t, file2, destData2)
+}
+
+func TestListDefaultContext(t *testing.T) {
+	meta := testDefaultMetadata()
```

Comment:
All the following tests have the same initialization, can you factorize them using a table test pattern?

Reply:
Indeed. Done